WorldTest.java

```diff
@@ -0,0 +1,149 @@
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class WorldTest {
+
+	@Test
+	/*
+	 * This test assures the create result for an empty string
+	 */
+	public void emptyWorldTest() {
+		// Create a new empty world
+		World w = new World(0, 2, 2);
+
+		// Passing the empty world to the toString() method should result in a
+		// string consisting of
+		// a few spaces and a new line
+		String s = "  " + "\n";
+		assertEquals(s, w.toString());
+	}
+
+	@Test
+	/*
+	 * This test is checking that the header numbering is consistent each time a
+	 * world of size 10x10 is constructed
+	 */
+	public void checkHeaderFormat() {
+		// Create a new world
+		World w = new World(10, 0, 0);
+
+		// The numbering format across the header should be from 0-9
+		String s = "  " + "0123456789\n";
+		assertTrue(w.toString().contains(s));
+	}
+
+	@Test
+	/*
+	 * This test is checking that the row count is consistent each time a world
+	 * of size 10x10 is constructed
+	 */
+	public void checkRowFormat() {
+		// Create a new world 10x10
+		World w = new World(10, 0, 0);
+
+		// There should consist 11 rows due to the extra newline placed after
+		// the game board
+		String[] rows = w.toString().split("\n");
+		assertEquals(rows.length, 11);
+	}
+
+	@Test
+	/*
+	 * This test checks whether for the parameters that produce the same random
+	 * number, the correct output string is produced each time on one iteration
+	 */
+	public void correctOutputOneIteration() {
+		// Create a new world
+		World w = new World(5, 123456789, 50);
+
+		// Passing the world to the toString() method should result in a string
+		// consisting of
+		// a one X in row 0, three X's in row 1, one X in row 3, and three X's
+		// in row 4
+		String s = "  01234\n0 .X.X.\n1 .XX..\n2 X...X\n3 ...X.\n4 X.X..\n";
+		assertEquals(s, w.toString());
+	}
+
+	@Test
+	/*
+	 * By passing in the specific arguments 2, 3, 50 The same string is
+	 * constructed each time
+	 */
+	public void reptivitiveIterationsTest() {
+		// Create a new world
+		World w = new World(2, 3, 0);
+
+		// Passing the world to the toString() method should result in a
+		// matrix with only periods and no 'Xs'
+		// Calling w.iterate() in a for loop 100 times ensures this is the case
+		// each time
+		for (int i = 0; i < 100; i++) {
```

Comment:
Doing this is mostly testing the iterate() method instead of toString(), especially so many times.  A valid although mostly superfluous test for this assignment.